common/replication/block_fetcher.go

```diff
@@ -475,6 +476,121 @@ func (bf *BlockFetcher) setup() {
 	}()
 
 	bf.suspects = suspectSet{max: bf.FetcherConfig.MaxByzantineNodes}
+
+	/* If the `VerifyBlock` is not set, BlockFetcher tries to pull genesis block and intialize the `VerifyBlock` function
+	based on it. It will try to pull genesis block(block number 0) till it suceeds.*/
+	if bf.VerifyBlock == nil {
+		bf.Logger.Debugf("VerifyBlock Function is not set.")
+		bf.setupVerifierInProgress = true
+		for bf.setupVerifierInProgress {
+			// keep polling till we get a genesis block.
+			ok := bf.setUpVerifierFromGenesisBlock()
+			bf.Logger.Debugf("Try to pull genesis block %v", ok)
+			if ok {
+				bf.Logger.Debugf("Successfully pulled a genesis block.")
+				bf.setupVerifierInProgress = false
+			}
+		}
+	}
+}
+
+// setUpVerifierFromGenesisBlock connects to orderers and tries to pull genesis block from f+1 distinct
+// orderers and sets up verifier
+func (bf *BlockFetcher) setUpVerifierFromGenesisBlock() bool {
+	candidates := bf.blockSourceCandidates()
+
+	var wg sync.WaitGroup
+	wg.Add(len(candidates))
+	var blockSources []BlockSource
```

Comment:
why do you need `blockSources` ? You append to it but never use it later.

Reply:
not needed. removed.